extension/httpfs/httpfs.cpp

```diff
@@ -515,22 +514,23 @@ void HTTPFileSystem::Read(FileHandle &handle, void *buffer, int64_t nr_bytes, id
 	bool skip_buffer = hfh.flags.DirectIO() || hfh.flags.RequireParallelAccess();
 	if (skip_buffer && to_read > 0) {
 		GetRangeRequest(hfh, hfh.path, {}, location, (char *)buffer, to_read);
+
+		std::lock_guard<std::mutex> lck(hfh.mu);
```

Comment:
Can we only grab the lock when `RequireParallelAccess()`? i don't think we should grab the lock when only `DirectIO()` is true, right?